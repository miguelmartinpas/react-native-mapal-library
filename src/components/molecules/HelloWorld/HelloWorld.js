import React from 'react';
import styled from 'styled-components/native';

const Container = styled.View`
  flex: 1;
  background-color: ${props => props.backgroundColor};
  justify-content: center;
  align-items: center;
`;

const Title = styled.Text`
  font-size: 24px;
  font-weight: 500;
  color: palevioletred;
`;

const HelloWorld = () => {
  return (
    <Container backgroundColor="papayawhip">
      <Title>Hello World</Title>
    </Container>
  );
};

export default HelloWorld;
