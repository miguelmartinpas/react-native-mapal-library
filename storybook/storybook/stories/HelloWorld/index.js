import React from 'react';
import { HelloWorld } from '../../../src';

export default () => {
  return <HelloWorld />;
};
