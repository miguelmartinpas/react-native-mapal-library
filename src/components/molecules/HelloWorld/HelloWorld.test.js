import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import HelloWorld from './index';

describe('HelloWorld', () => {
  describe('Rendering', () => {
    it('renders correctly', () => {
      const tree = renderer.create(<HelloWorld />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders correctly', () => {
      const tree = renderer.create(<HelloWorld />).getInstance();
      expect(tree).toBeDefined();
    });

    it('should match to snapshot', () => {
      const component = shallow(<HelloWorld />);
      expect(component).toMatchSnapshot();
    });
  });
});
