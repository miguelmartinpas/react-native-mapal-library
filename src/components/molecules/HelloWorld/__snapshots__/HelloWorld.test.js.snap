// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld Rendering renders correctly 1`] = `
<View
  backgroundColor="papayawhip"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "papayawhip",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "palevioletred",
          "fontSize": 24,
          "fontWeight": "500",
        },
      ]
    }
  >
    Hello World
  </Text>
</View>
`;

exports[`HelloWorld Rendering should match to snapshot 1`] = `ShallowWrapper {}`;
